### DISCUSSION TITLE
Upload image with compression

### DISCUSSION THREAD

**Discussion by tapeo** [photoprism/photoprism#1916] [Ideas]
Hi!

Is it possible to upload pictures using the web app with compression. Could be an option for the people who doesn't care about image quality and instead needs to save disk space.

Thanks!

**Comment by lastzero** [comment]
What compression? You mean lower resolution? A different format?

Browsers typically don't compress uploaded files as they start the request, so they can't know if the server supports compression. Might be possible to hack something with JS, but is it really worth it? Could be unstable and slow. If we have an awful lot of time, we can try WebAssembly.

**Reply by tapeo** [reply]
Yes! I mean something to reduce the picture quality before send it to the server to save disk space! :)

**Reply by lastzero** [reply]
It would be possible with a native app or via WebAssembly in the browser:

- https://silvia-odwyer.github.io/photon/
- https://github.com/nodeca/pica

We already have more than enough work, so it won't happen in the short term. Unless someone sends us a really big check! ;)

**Reply by nautatva** [reply]
I had earlier worked on a face-recog algorithm and just compressing a photo would do a major hit on accuracy.

**Comment by mmolotov** [comment]
hi @lastzero, what about image compression on the backend?
this could be after the indexing task: images are uploaded at their original size, then indexed, then OPTIONALLY compressed to save space.
